sklearn/utils/tests/test_validation.py

```diff
@@ -1445,3 +1446,31 @@ def test_check_array_deprecated_matrix():
     )
     with pytest.warns(FutureWarning, match=msg):
         check_array(X)
+
+
+@pytest.mark.parametrize("names", [[f"col_{i}" for i in range(3)], list(range(3))])
+def test_get_feature_names_pandas(names):
+    """Get feature names with pandas dataframes."""
+    pd = pytest.importorskip("pandas")
+    X = pd.DataFrame([[1, 2, 3], [4, 5, 6]], columns=names)
+
+    names = _get_feature_names(X)
+    assert_array_equal(names, names)
+
```

Comment:
We should also cover the case where names are not passed explicitly and we get a `RangeIndex` as a result.

```python
>>> import pandas as pd, numpy as np
>>> pd.DataFrame(np.random.normal(size=(5, 3))).columns
RangeIndex(start=0, stop=3, step=1)
>>> pd.DataFrame(np.random.normal(size=(5, 3)), columns=None).columns
RangeIndex(start=0, stop=3, step=1)
```

This is slightly different than  the `list(range(3))` case currently covered in this test.

Reply:
yeah this is an important case for sure.

Reply:
The PR has been updated and this comment is no longer relevant.